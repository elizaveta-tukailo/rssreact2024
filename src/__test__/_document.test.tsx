import { render } from '@testing-library/react';
import Document, { Html, Head, Main, NextScript } from 'next/document';
import '@testing-library/jest-dom';

class MyDocument extends Document {
  render() {
    return (
      <Html>
        <Head>
          <link rel="icon" href="/app/favicon.ico" />
        </Head>
        <body>
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument;

const TestDocument = () => (
  <html>
    <head>
      <link rel="icon" href="/app/favicon.ico" />
    </head>
    <body>
      <div id="__next">
        <main />
        <script />
      </div>
    </body>
  </html>
);

describe('MyDocument', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<TestDocument />);
    expect(asFragment()).toMatchSnapshot();
  });
});
