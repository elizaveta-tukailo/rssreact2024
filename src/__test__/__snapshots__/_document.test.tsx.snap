// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document html Should render correctly 1`] = `
<DocumentFragment>
  <link
    href="/app/favicon.ico"
    rel="icon"
  />
  <div
    id="__next"
  >
    <main />
    <script />
  </div>
</DocumentFragment>
`;

exports[`MyDocument renders correctly 1`] = `
<DocumentFragment>
  <link
    href="/app/favicon.ico"
    rel="icon"
  />
  <div
    id="__next"
  >
    <main />
    <script />
  </div>
</DocumentFragment>
`;

exports[`Test document hrml Check if document renders correctly 1`] = `
<DocumentFragment>
  <link
    href="/app/favicon.ico"
    rel="icon"
  />
  <div
    id="__next"
  >
    <main />
    <script />
  </div>
</DocumentFragment>
`;
